lib/defaults.js

```diff
@@ -28,7 +28,7 @@
 
 	const cacheExtra = process.platform === 'win32' ? 'npm-cache' : '.npm';
 	const cacheRoot = process.platform === 'win32' ? process.env.APPDATA : home;
-	const cache = path.resolve(cacheRoot, cacheExtra);
+	const cache = cacheRoot ? path.resolve(cacheRoot, cacheExtra) : undefined;
```

Comment:
I think `cacheRoot` should fall back to `home` if `APPDATA` isn't available. See https://github.com/npm/npm/blob/latest/lib/config/defaults.js#L85. Could you change it accordingly?

Reply:
updated. thanks